src/cluster.c

```diff
@@ -4942,6 +4964,146 @@ void addNodeReplyForClusterSlot(client *c, clusterNode *node, int start_slot, in
     setDeferredArrayLen(c, nested_replylen, nested_elements);
 }
 
+/* Add detailed information of a node to the output buffer of the given client. */
+void addNodeDetailsToShardReply(client *c, clusterNode *node) {
+    int reply_count = 0;
+    void *node_replylen = addReplyDeferredLen(c);
+    addReplyBulkCString(c, "id");
+    addReplyBulkCBuffer(c, node->name, CLUSTER_NAMELEN);
+    reply_count++;
+
+    int port = server.cluster_announce_port ? server.cluster_announce_port : server.port;
+    if (port) {
+        addReplyBulkCString(c, "port");
+        addReplyLongLong(c, node->port);
+        reply_count++;
+    }
+
+    int tls_port = server.cluster_announce_tls_port ? server.cluster_announce_tls_port : server.tls_port;
+    if (tls_port) {
+        addReplyBulkCString(c, "tls-port");
+        addReplyLongLong(c, tls_port);
```

Comment:
please take a look here (tls_port and the above port in tls mode)
the cluster-shards test failed in tls. (https://github.com/redis/redis/runs/5578694306?check_suite_focus=true#step:9:682)

with this patch, the test pass, but I'm not very familiar with tls mode
```
    int port = server.cluster_announce_port ? server.cluster_announce_port : server.port;
    int tls_port = server.cluster_announce_tls_port ? server.cluster_announce_tls_port : server.tls_port;
    if (port) {
        addReplyBulkCString(c, "port");
        if (tls_port)
            addReplyLongLong(c, node->pport);
        else
            addReplyLongLong(c, node->port);
        reply_count++;
    }

    if (tls_port) {
        addReplyBulkCString(c, "tls-port");
        addReplyLongLong(c, node->port);
        reply_count++;
    }
```

in case anyone will fix it, i would like to mention:
1. since we deprecated cluster-slots, maybe we need to add a doc_flags in cluster-slots.json like others
2. i notice cluster-shards.json, it's indented by two spaces, we use four spaces elsewhere

Reply:
For 1, we marked it as deprecated. Was there something else that you meant?

Reply:
i mean the doc_flags, like other commands?
```
        "deprecated_since": "6.2.0",
        "replaced_by": "`GEOSEARCH` and `GEOSEARCHSTORE` with the `BYRADIUS` argument",
        "doc_flags": [
            "DEPRECATED"
        ],
```

Reply:
Oh, I guess I didn't realize that was a thing.

Reply:
@hpatro Do you have time to address this? Otherwise I can take a look tomorrow.

Reply:
@enjoy-binbin Regarding the `port` and `tls-port` logic: `server.tls_port` is always a TLS port and `server.port` is always a plaintext (AKA cleartext) port if specified. For the cluster bus, it depends on the `server.tls_cluster` variable:

```C
if (server.tls_cluster) {
    // node->port is TLS port (always set)
    // node->pport is plaintext port, optional (if non-zero)
} else {
    // node->port is plaintext port (always set)
    // There is no TLS port
}
```